### ISSUE TITLE
pdns_auth: mdb_env_set_maxreaders config option

### ISSUE THREAD

**Issue by Algirdas-Z** [PowerDNS/pdns#13418]
- Program: Authoritative 
 - Issue type: Feature request

### Short description

Add configuration option into config file to be able to change value of `mdb_env_set_maxreaders` when using LMDB backend

### Usecase

In general use there is no issues with, yet when pushing to high number of threads of pdns, looks like pdns gets blocked by locks

```
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
pdns_server[218772]: Exception building answer packet for domain.com/A (Unable to start RO transaction: MDB_READERS_FULL: Environment maxreaders limit reached) sending out servfail
```

### Description

This could be one of the tunables that can help in high performance scenarios, since powerdns doesn't have this configurable at the moment, can't test it and see if it actually results in better performance

**Comment by Habbie** [comment]
I thought this would be a simple change, similar to how we handle mapsize. Instead I found out we currently don't do mapsize right for the LMDB shards, so it'll need some refactoring.

**Comment by miodvallat** [comment]
Note that the mapsize handling has been fixed in #15677, so all you need is to rebase your branch and it should be ready to go in (with some documentation bits).

**Comment by Algirdas-Z** [comment]
@miodvallat how one should change/define such options then?

**Comment by miodvallat** [comment]
At the moment, the ability to configure a non-default value for maxreaders is only available in @Habbie's [lmdb-maxreaders-unfinished](https://github.com/Habbie/pdns/tree/lmdb-maxreaders-unfinished) branch which is not merged yet.

**Comment by Algirdas-Z** [comment]
> At the moment, the ability to configure a non-default value for maxreaders is only available in [@Habbie](https://github.com/Habbie)'s [lmdb-maxreaders-unfinished](https://github.com/Habbie/pdns/tree/lmdb-maxreaders-unfinished) branch which is not merged yet.

Right, thanks for the update, will keep an eye on this progress, when using modern AMD CPUs its quite obvious we need more than 128 readers (at least from my testing)
